apps/ledger-live-mobile/src/families/ethereum/EthereumCustomFees/Ethereum1559CustomFees.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState, memo, useMemo, useCallback } from "react";
+import { formatCurrencyUnit } from "@ledgerhq/live-common/currencies/formatCurrencyUnit";
+import useBridgeTransaction from "@ledgerhq/live-common/bridge/useBridgeTransaction";
+import { getGasLimit } from "@ledgerhq/live-common/families/ethereum/transaction";
+import { useTheme } from "@react-navigation/native";
+import { Transaction } from "@ledgerhq/live-common/families/ethereum/types";
+import { getMainAccount } from "@ledgerhq/live-common/account/helpers";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { StyleSheet, View } from "react-native";
+import { BigNumber } from "bignumber.js";
+import { Trans } from "react-i18next";
+import invariant from "invariant";
+import LText from "../../../components/LText";
+import Button from "../../../components/Button";
+import EthereumGasLimit from "../SendRowGasLimit";
+import EditFeeUnitEthereum from "../EditFeeUnitEthereum";
+import SectionSeparator from "../../../components/SectionSeparator";
+import { inferMaxFeeRange, inferMaxPriorityFeeRange } from "./utils";
+
+type Props = {
+  account: AccountLike;
+  parentAccount: Account | null | undefined;
+  transaction: Transaction;
+  onValidateFees: (transaction: Partial<Transaction>) => () => void;
+};
+
+const Ethereum1559CustomFees = ({
+  account,
+  parentAccount,
+  onValidateFees,
+  transaction: originalTransaction,
+}: Props) => {
+  const { colors } = useTheme();
+
+  invariant(originalTransaction.family === "ethereum", "not ethereum family");
+  invariant(account, "no account found");
+
+  const { currency } = getMainAccount(account, parentAccount);
+  const feeUnit =
+    currency.units.length > 1 ? currency.units[1] : currency.units[0];
```

Comment:
it could be wise to externalize this logic somewhere else, like "getDefaultFeeUnit" on ethereum helper. because it's weird why [1] and [0] (and i think we may have this in multiple places)